From 007f2c503ecc85b9ecfa5d24256986a830857f02 Mon Sep 17 00:00:00 2001
From: Janne Grunau <j@jannau.net>
Date: Sun, 11 Dec 2022 20:08:45 +0100
Subject: [PATCH 38/53] HID: transport: spi: Check status message after
 transmits

Probably pointless but might be helpful to debug issues.
Gets rid of 'spi_hid_apple_status_ok' is unused compiler warning.

Signed-off-by: Janne Grunau <j@jannau.net>
---
 drivers/hid/spi-hid/spi-hid-apple-core.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/hid/spi-hid/spi-hid-apple-core.c b/drivers/hid/spi-hid/spi-hid-apple-core.c
index 67c70edbe..a10a5a38b 100644
--- a/drivers/hid/spi-hid/spi-hid-apple-core.c
+++ b/drivers/hid/spi-hid/spi-hid-apple-core.c
@@ -248,7 +248,17 @@ static int spihid_apple_request(struct spihid_apple *spihid, u8 target, u8 unk0,
 	pkt->crc16 = cpu_to_le16(crc16(0, spihid->tx_buf,
 				 offsetof(struct spihid_transfer_packet, crc16)));
 
+	memset(spihid->status_buf, 0, sizeof(spi_hid_apple_status_ok));
+
 	err = spi_sync(spihid->spidev, &spihid->tx_msg);
+
+	if (memcmp(spihid->status_buf, spi_hid_apple_status_ok,
+		   sizeof(spi_hid_apple_status_ok))) {
+		u8 *b = spihid->status_buf;
+		dev_warn_ratelimited(&spihid->spidev->dev, "status message "
+				     "mismatch: %02x %02x %02x %02x\n",
+				     b[0], b[1], b[2], b[3]);
+	}
 	mutex_unlock(&spihid->tx_lock);
 	if (err < 0)
 		return err;
-- 
2.50.1

